### DISCUSSION TITLE
strimzi operator operator supports sasl.enabled.mechanisms=PLAIN

### DISCUSSION THREAD

**Discussion by lanzhiwang** [strimzi/strimzi-kafka-operator#4466] [General]
Due to historical business reasons, some of our customers use `SASL PLAIN` when communicating with Kafka, so we want to change the `SASL SCRAM-SHA-512` currently supported by the operator to `SASL PLAIN`.

I took a cursory look at the code, if I use a simpler method, I only need to modify some constants in the `api/` `cluster-operator/` `docker-images/` directory.

Would you like to ask about this idea?

**Comment by scholzj** [comment]
I do not think it is so simple. SASL PLAIN is using usernames and passwords stored in plaintext file. So you would also need to maintain and mount this file. This is different for SCRAM-SHA which stores the usernames and the passwords (or, well, what ever is left of them after the hashing) in Zookeeper. So adding SASL PLAIN support would IMHO be more complicated than just changing some constants. 

SASL PLAIN being insecure and rather impractical is one of the reasons we do not support it and have no plans to add it right now.
